code/modules/mob/living/carbon/carbon.dm

```diff
@@ -440,6 +440,9 @@
 	return ..()
 
 /mob/living/carbon/proc/vomit(var/lost_nutrition = 10, var/blood = 0, var/stun = 1, var/distance = 0, var/message = 1, var/toxic = 0)
+	if(dna && dna.species && !(NOHUNGER in dna.species.specflags))
+		return 1

```

Comment:
Forgive my ignorance if I'm wrong, won't this cause all things WITHOUT "NOHUNGER" to NOT throw up?